rocketpy/Flight.py

```diff
@@ -1596,190 +1612,216 @@ def uDotParachute(self, t, u, postProcessing=False):
             "Time (s)",
-            "Wind Velocity Y (North) (m/s)",
-            interpolation,
+            "e2",
+            interpolation="spline",
+            extrapolation="natural",
         )
-        self.density = Function(
-            self.density, "Time (s)", "Density (kg/m³)", interpolation
+
+    @cached_property
+    def e3(self):
+        return Function(
+            np.array(self.solution)[:, [0, 10]],
+            "Time (s)",
+            "e3",
+            interpolation="spline",
+            extrapolation="natural",
         )
-        self.pressure = Function(
-            self.pressure, "Time (s)", "Pressure (Pa)", interpolation
+
+    @cached_property
+    def w1(self):
+        return Function(
+            np.array(self.solution)[:, [0, 11]],
+            "Time (s)",
+            "ω1 (rad/s)",
+            interpolation="spline",
+            extrapolation="natural",
         )
-        self.dynamicViscosity = Function(
-            self.dynamicViscosity, "Time (s)", "Dynamic Viscosity (Pa s)", interpolation
+
+    @cached_property
+    def w2(self):
+        return Function(
+            np.array(self.solution)[:, [0, 12]],
+            "Time (s)",
+            "ω2 (rad/s)",
+            interpolation="spline",
+            extrapolation="natural",
         )
-        self.speedOfSound = Function(
-            self.speedOfSound, "Time (s)", "Speed of Sound (m/s)", interpolation
+
+    @cached_property
+    def w3(self):
+        return Function(
+            np.array(self.solution)[:, [0, 13]],
+            "Time (s)",
+            "ω3 (rad/s)",
+            interpolation="spline",
+            extrapolation="natural",
         )
 
-        # Process fourth type of output - values calculated from previous outputs
+    # Process fourth type of output - values calculated from previous outputs
 
-        # Kinematics functions and values
-        # Velocity Magnitude
-        self.speed = (self.vx**2 + self.vy**2 + self.vz**2) ** 0.5
-        self.speed.setOutputs("Speed - Velocity Magnitude (m/s)")
+    # Kinematics functions and values
+    # Velocity Magnitude
+    @cached_property
+    def speed(self):
+        speed = (self.vx**2 + self.vy**2 + self.vz**2) ** 0.5
+        speed.setOutputs("Speed - Velocity Magnitude (m/s)")
+        return speed
+
+    @cached_property
+    def maxSpeedTime(self):
         maxSpeedTimeIndex = np.argmax(self.speed[:, 1])
-        self.maxSpeed = self.speed[maxSpeedTimeIndex, 1]
-        self.maxSpeedTime = self.speed[maxSpeedTimeIndex, 0]
-        # Acceleration
-        self.acceleration = (self.ax**2 + self.ay**2 + self.az**2) ** 0.5
-        self.acceleration.setOutputs("Acceleration Magnitude (m/s²)")
+        return self.speed[maxSpeedTimeIndex, 0]
+
+    @cached_property
+    def maxSpeed(self):
+        return self.speed(self.maxSpeedTime)
+
+    # Accelerations
+    @cached_property
+    def acceleration(self):
+        if self.retrievedAcceleration is not True:
```

Comment:
what is the purpose of using this condition when you already have the @cached_property in place?

Reply:
Thank you for the review, it pushed me to get rid of those booleans, I'm adding new changes and asking a re-review